app/controllers/state_file/questions/az_qualifying_organization_contributions_controller.rb

```diff
@@ -0,0 +1,74 @@
+module StateFile
+  module Questions
+    class AzQualifyingOrganizationContributionsController < QuestionsController
+      include ReturnToReviewConcern
+
+      before_action -> { @filing_year = Rails.configuration.statefile_current_tax_year }
+
+      def index
+        @contribution_count = contributions.count
+        redirect_to action: :new unless contributions.present?
+      end
+
+      def edit
+        contribution.made_contributions = "yes"
+      end
+
+      def update
+        contribution.assign_attributes(az321_contribution_params)
+
+        if contribution.save
+          redirect_to action: :index, return_to_review: params[:return_to_review]
+        else
+          render :edit
+        end
+      end
+
+      def new 
+        @contribution = contributions.build
+      end
+
+      def create
+        @contribution = contributions.build(az321_contribution_params)
+
+        return redirect_to next_path if @contribution.made_contributions == "no"
+
+        if @contribution.save(context: :form_create)
+          redirect_to action: :index, return_to_review: params[:return_to_review]
```

Comment:
`redirect_to action: :index` is the same as `redirect_to next_path`, no? also, are we intentionally not carrying over the `:return_to_review` param on line 34? i would think we'd want it passed through either way but maybe i'm missing something

Reply:
ok! i think i don't totally understand why it changes but i believe you. any thoughts on the `:return_to_review` param?

Reply:
The way the flow currently works is something like:

```mermaid
sequenceDiagram
    AZ Review Page->>Form Index: User clicks to review
    Form Index->>Form Edit: User clicks 'Add another' or 'Edit'
    Form Edit->>Form Index: User saves
    Form Index->>AZ Review Page: User clicks continue, using next_path underneath
```

Reply:
hm, ok, the scenario i think is possible here:
1. get to the index from the review page with review param
2. click to add a new contribution
3. select “no” they didn’t make a contribution and click continue to get back to the index, losing review param (line 34)
4. click continue on index
5. instead of going back to the review page, they will end up on whatever the next intake question is

i might be wrong though. probably best to just test it out manually because thinking about it is confusing me

Reply:
Yea, that case does seem not to work. Also, it looks like I neglected to add a panel to the AzReviewController, so will work on that together with this fix and will come back for another review.